components/buildless-serverless/internal/controller/state/deployment_status.go

```diff
@@ -0,0 +1,126 @@
+package state
+
+import (
+	"context"
+	"fmt"
+	serverlessv1alpha2 "github.com/kyma-project/serverless/api/v1alpha2"
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func sFnDeploymentStatus(ctx context.Context, m *stateMachine) (stateFn, *ctrl.Result, error) {
+	deploymentName := NewDeploymentBuilder(m).deploymentName()
+	deployment := appsv1.Deployment{}
+	m.client.Get(ctx, client.ObjectKey{
+		Namespace: m.state.instance.Namespace,
+		Name:      deploymentName,
+	}, &deployment)
+
+	// ready deployment
+	if isDeploymentReady(deployment) {
+		m.log.Info(fmt.Sprintf("deployment %s ready", deploymentName))
+
+		m.state.instance.UpdateCondition(
+			serverlessv1alpha2.ConditionRunning,
+			metav1.ConditionTrue,
+			serverlessv1alpha2.ConditionReasonDeploymentReady,
+			fmt.Sprintf("Deployment %s is ready", deploymentName))
+
+		return requeueAfter(m.functionConfig.FunctionReadyRequeueDuration)
+	}
+
+	// unhealthy deployment
+	if hasDeploymentConditionFalseStatusWithReason(deployment.Status.Conditions, appsv1.DeploymentAvailable, MinimumReplicasUnavailable) {
+		m.log.Info(fmt.Sprintf("deployment unhealthy: %q", deploymentName))
+
+		m.state.instance.UpdateCondition(
+			serverlessv1alpha2.ConditionRunning,
+			metav1.ConditionUnknown,
+			serverlessv1alpha2.ConditionReasonMinReplicasNotAvailable,
+			fmt.Sprintf("Minimum replicas not available for deployment %s", deploymentName))
+
+		return stop()
+	}
+
+	// deployment not ready
+	if hasDeploymentConditionTrueStatus(deployment.Status.Conditions, appsv1.DeploymentProgressing) {
+		m.log.Info(fmt.Sprintf("deployment %q not ready", deploymentName))
+
+		m.state.instance.UpdateCondition(
+			serverlessv1alpha2.ConditionRunning,
+			metav1.ConditionUnknown,
+			serverlessv1alpha2.ConditionReasonDeploymentWaiting,
+			fmt.Sprintf("Deployment %s is not ready yet", deploymentName))
+
+		return stop()
```

Comment:
should it be requeued instead of stopped? 🤔 I'm not sure, but I think we expect to check if deployment is ready after a specific time

Reply:
You are right. This fragment is based on old serverless where the default return is requeue after one second.